.github/workflows/haskell.yml

```diff
@@ -31,14 +30,22 @@ jobs:
         path: ~/.cabal
         key: ${{ runner.OS }}-${{ matrix.ghc }}-cabal-0
 
-    - name: Run ShellCheck
-      uses: ludeeus/action-shellcheck@master
-      if: matrix.os == 'ubuntu-latest'
-      with:
-        ignore: tests
-
     - run: cabal update
     - name: Build
       run: cabal build
     - name: Test
       run: cabal test
+    - name: Check
+      run: cabal check
+    - name: Generate sdist
+      run: cabal sdist
+    - uses: actions/upload-artifact@v2
+      with:
+        path: dist-newstyle/sdist/hie-bios*.tar.gz
```

Comment:
what haskell-ci does is to use cabal build, cabal install and cabal test on the sdist tar archive. This way it detects that a build really works after it is uploaded to hackage.

Reply:
Yeah, I think its best to leave that to the travis script, since there seemed to be a lot of `sed`ing needed to get that to work. We can use this just for build artefacts of the sdist